orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/monitoreddeploy/EvaluateDeploymentHealthTask.java

```diff
@@ -0,0 +1,105 @@
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.deploymentmonitor.models.EvaluateHealthRequest;
+import com.netflix.spinnaker.orca.deploymentmonitor.models.EvaluateHealthResponse;
+import com.netflix.spinnaker.orca.deploymentmonitor.models.StatusReason;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+@Component
+@ConditionalOnProperty(value = "monitored-deploy.enabled")
+public class EvaluateDeploymentHealthTask extends MonitoredDeployBaseTask {
+  @Autowired
+  EvaluateDeploymentHealthTask(
+      DeploymentMonitorServiceProvider deploymentMonitorServiceProvider, Registry registry) {
+    super(deploymentMonitorServiceProvider, registry);
+  }
+
+  @Override
+  public TaskResult executeInternal() {
+    EvaluateHealthRequest request = new EvaluateHealthRequest(stage);
+
+    log.info(
+        "Evaluating health of deployment {} at {}% with {}",
+        request.getNewServerGroup(), request.getCurrentProgress(), monitorDefinition);
+
+    EvaluateHealthResponse response = monitorDefinition.getService().evaluateHealth(request);
+    sanitizeAndLogResponse(response);
+
+    EvaluateHealthResponse.NextStepDirective directive = response.getNextStep().getDirective();
+
+    Id statusCounterId =
+        registry
+            .createId("deploymentMonitor.healthStatus")
+            .withTag("monitorId", monitorDefinition.getId())
+            .withTag("status", directive.toString());
+
+    registry.counter(statusCounterId).increment();
+
+    if (directive != EvaluateHealthResponse.NextStepDirective.WAIT) {
+      // Log timing
+      Id timerId =
+          registry
+              .createId("deploymentMonitor.timing")
+              .withTag("monitorId", monitorDefinition.getId())
+              .withTag("status", directive.toString());
+
+      double duration = Instant.now().toEpochMilli() - stage.getStartTime();
+      registry.gauge(timerId).set(duration);
+    }
+
+    List<StatusReason> statusReasons =
+        Optional.ofNullable(response.getStatusReasons()).orElse(Collections.emptyList());
+
+    TaskResult.TaskResultBuilder taskResultBuilder;
+
+    switch (directive) {
+        // TODO(mvulfson): Actually implement this case in the stages
+      case COMPLETE:
+        taskResultBuilder =
+            TaskResult.builder(ExecutionStatus.SUCCEEDED).output("skipToPercentage", 100);
+        break;
+
+      case CONTINUE:
+        taskResultBuilder = TaskResult.builder(ExecutionStatus.SUCCEEDED);
+        break;
+
+      case WAIT:
+        taskResultBuilder = TaskResult.builder(ExecutionStatus.RUNNING);
+        break;
+
+      case ABORT:
+      default:
```

Comment:
if we flow into the default, sounds like we want to throw some kind of exception. Seems bad!

Reply:
well, this is just to appease the compiler, since in the `sanitizeAndLog` will spit out an 
 error in this case and then coorce the `directive` into `ABORT` - weird? https://github.com/spinnaker/orca/pull/3128/files/982f910d24775f7304bee9183bb7f0aa95c88909#diff-f11fb77f23e56127050a6b16bbbc272eR53

Reply:
resolved offline